core_contracts/loans/loans/loans.py

```diff
@@ -724,23 +756,83 @@ def retireRedeem(self, _symbol: str, _sicx_from_lenders: int) -> None:
 
             remaining_supply -= user_debt
         price = asset.priceInLoop()
-        self.AssetRetired(self.msg.sender, _symbol, _redeemed, price, _redeemed,
+        self.AssetRetired(self.msg.sender, _symbol, _redeemed, price,
                           total_batch_debt, str(redeemed_dict))
 
+    @external
+    @only_rebalance
+    def generateBnusd(self, _tokens_to_retire: int) -> None:
+        """
+        This function will  add off debt to a batch of
+        borrowers proportionately along with their collateral.
+        :param _tokens_to_retire: Max bnUSD token to retire.
+        :type _tokens_to_retire: int
+        """
+        _symbol = "sICX"
+        _from = self._rebalance.get()
+        asset = self._assets[_symbol]
+
+        if not (asset and asset.is_active()):
```

Comment:
We can remove this for now since there is no chance this is anything other than sICX.